### ISSUE TITLE
Close settings dialog disables menus

### ISSUE THREAD

**Issue by Artiom-M** [ribtoks/xpiks#289]
I came across a bug with UI:
1. Open "Settings" dialog 
2. Close is with "Close" button

Result: all menu items are disabled except for Settings, About and Exit

**Comment by Ribtoks** [comment]
What is "properties"?

**Comment by Artiom-M** [comment]
I mean "Settings"

**Comment by Ribtoks** [comment]
Does not repro in Windows.

A lot of menu items are disabled by default, when there are no artworks added to Xpiks. Some items get enabled only when artwork is selected (like Zip, Reimport and others).

Please write clean steps to reproduce e.g. add 1 items, open settings, close settings - then this particular menu items is disabled. Your description is not clear at all.

**Comment by Artiom-M** [comment]
All steps are in the report. I add here screenshots before and after:
http://s32.postimg.org/6tg4dapl1/Screenshot_from_2016_05_05_16_30_11.png
http://s32.postimg.org/ui0o5osg5/Screenshot_from_2016_05_05_16_31_02.png

**Comment by Ribtoks** [comment]
Tried in OS X and Windows. Does not repro. I think it's a bug of your distro's Qt.
Will try in openSUSE as well. If it wouldn't repro, will close the bug as "wontfix", because I don't care to hotfix all buggy Qt distributions over the all Linux distro.

**Comment by Ribtoks** [comment]
The problem is that in SettingsWindow dialogDestruction() signal is not getting called, but it should. 

    signal dialogDestruction();

    onClosing: dialogDestruction();

    function closeSettings() {
        settingsWindow.close()
    }

**Comment by Ribtoks** [comment]
Just tried in openSUSE 13.2
It does not repro.

**Comment by Artiom-M** [comment]
Do you try last code base?

**Comment by Ribtoks** [comment]
Yes. Fresh pull.
It doesn't matter since I didn't touch that area for a very long time. Even
not the latest code base will work.
5 трав. 2016 р. 20:05 "Artiom-M" <notifications@github.com> пише:

**Comment by Artiom-M** [comment]
In my Suse it works as well. And I noticed that when I open _Settings_ on Suse the menus on the background are greyed out. While on debian menus are greyed out the same way, but only after _Settings_ dialog is closed. Could you point me where this behavior is in the code?

**Comment by Ribtoks** [comment]
Menus are greyed out if `applicationWindow.openedDialogsCount > 1` (main.qml). When a popup dialog is opened, `openedDialogsCount` is increased. When dialogDestruction() signal is emited, count is decremented. Everything is orchestrated in Common.js in `launchDialog`.

**Comment by Artiom-M** [comment]
Yes, this is QT but, but I found a fix:
add _dialogDestruction();_ into  function _closeSettings()_  just after a call to _close()_ for this window,.

**Comment by Ribtoks** [comment]
Then we will need to remove `dialogDestruction()` call from `onClosing`. I will test it later if it works well. Don't have enough time today.

**Comment by Artiom-M** [comment]
Yes, it worked for Suse. I already posted a .deb without fix. It can work on Ubuntu if it's QT >=5.4, just wanted to confirm my solution. Now I can create a fixed package.

**Comment by Artiom-M** [comment]
File is updated in the folder I sent you

**Comment by Ribtoks** [comment]
Your fix will not work if user will close Settings Window using a system close button - this was the reason why I didn't put `dialogDestruction()` signal there by myself. So anyway package will be broken..

**Comment by Ribtoks** [comment]
A correct "fix" for buggy Qt in Debian in this case would be to remove `dialogDestruction()` signal at all and make SettingsWindow not modal dialog, but normal one.

**Comment by Artiom-M** [comment]
There is no system close button on any child window in xpiks on Linux.

------ Original Message ------

**Comment by Ribtoks** [comment]
But there's one in Windows and OS X. Current code in the repository is correct for any one of them (I mean sending `dialogDestruction()` in `onClosing`). It's only bug in current Debian's Qt distro. We should not create some crappy fix for that as a general case solution.

**Comment by Artiom-M** [comment]
Ok, I will create a patch for Debian, while waiting for distro upgrade

**Comment by Ribtoks** [comment]
Hm.. Maybe that's why the bug exists - because the window does not have a system close button. Can you try to comment `flags: Qt.Dialog` line in the Settings window and emit `dialogDestruction()` signal in `onClosing` (the same was as is now in the repo).

**Comment by Artiom-M** [comment]
So if I comment out `flags: Qt.Dialog` the close button is back and the signal is emitted. But still not with the "Close" button

**Comment by Ribtoks** [comment]
Feel free to patch debians package (have the system close button and additionally emit signal in "Close" button), but do not push your changes to the main repository which contains _correct_ implementation. I hope Debian's Qt will sort this out in close future.

**Comment by Artiom-M** [comment]
Next debian will have QT 5.5.1 but the release date is not known yet.

------ Original Message ------

**Comment by Artiom-M** [comment]
This works. Signal is emitted only if there is a system Close button.

**Comment by ribtoks** [comment]
Does not repro if Xpiks is built from vanilla Qt. Closing since Xpiks is now being shipped as AppImage and built from vanilla Qt.
